didcomm/unpack.py

```diff
@@ -0,0 +1,22 @@
+from didcomm.interfaces.did_resolver import DIDResolver
+from didcomm.interfaces.secrets_resolver import SecretsResolver
+from didcomm.types.message import Message
+from didcomm.types.mtc import MTC
+from didcomm.types.types import JSON
+from didcomm.types.unpack_result import UnpackResult, Metadata
+
+
+class Unpacker:
+
+    def __init__(self,
+                 mtc: MTC = None,
```

Comment:
It seems odd to take in an MTC object here. What is the intended effect? All messages unpacked by this unpacker instance must match the MTC? Messages unpacked with the unpacker inherit MTC?

Reply:
Agree that MTC is a confusing term here. Renamed it to `UnpackOpts`.
Passing `UnpackOpts` to `Unpacker`, the caller can control what consistency checks are done by `unpack` automatically, and how they are done, so that an error will be thrown/returned if a consistency check returns an error.
For example, the caller may want to be sure, that all received messages must be authenticated. He can send `expect_authenticated = True` for the `Unpacker`, and the Unpacker will throw an error if a message is not authenticated.
Please note, that this is an optional feature. Also in some cases the caller may not know in advance if the message must always be authenticated for example, as it may depend on the message type (be protocol specific). That's why we also return a `Metadata` as part of `unpack` result, so that the caller may do additional checks on App level if needed.